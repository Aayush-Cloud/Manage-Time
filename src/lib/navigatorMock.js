// @flow
export default {
  push: jest.fn(),
  pop: jest.fn(),
  popToRoot: jest.fn(),
  resetTo: jest.fn(),
  showModal: jest.fn(),
  dismissModal: jest.fn(),
  dismissAllModals: jest.fn(),
  showLightBox: jest.fn(),
  dismissLightBox: jest.fn(),
  handleDeepLink: jest.fn(),
  setOnNavigatorEvent: jest.fn(),
  setButtons: jest.fn(),
  setTitle: jest.fn(),
  toggleNavBar: jest.fn(),
};
