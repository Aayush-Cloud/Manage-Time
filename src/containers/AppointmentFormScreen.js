// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { reduxForm, submit } from 'redux-form';
import i18n from 'react-native-i18n';
import moment from 'moment';

import AppointmentForm from '../components/AppointmentForm';
import validate from '../lib/validateAppointmentForm';
import { WEEKLY } from '../lib/recurrence';
import { RECURRENCE_PICKER } from '../screens';
import { navBar } from '../themes';

export type AppointmentFormInput = AppointmentInput & {
  id?: string,
  course?: string,
};

type OwnProps = {
  index?: number,
  initialValues?: Object,
  onSubmit: (input: AppointmentFormInput, index: number | void) => void,
};

type Props = OwnProps & {
  dispatch: Function,
  navigator: NativeNavigator,
};

const DecoratedAppointmentForm: () => React.Node = reduxForm({
  form: 'appointmentForm',
  validate,
})(AppointmentForm);

// exported for tests
export class AppointmentFormScreen extends React.Component<Props, {}> {
  static navigatorStyle = navBar;
  static navigatorButtons = {
    rightButtons: [
      {
        id: 'save',
        title: i18n.t('save'),
      },
    ],
  };

  constructor(props: Props) {
    super(props);
    this.props.navigator.setOnNavigatorEvent(this._handleNavigatorEvent);
  }

  _handleNavigatorEvent = event => {
    const { dispatch } = this.props;
    if (event.type === 'NavBarButtonPress') {
      if (event.id === 'save') {
        dispatch(submit('appointmentForm'));
      }
    }
  };

  _getInitialValues = (currentTime: moment) => {
    const startOfHour = currentTime.clone().startOf('hour');
    const startOfDay = currentTime.clone().startOf('day');
    return {
      starttime: startOfHour,
      endtime: startOfHour.clone().add(1, 'hour'),
      startdate: startOfDay,
      enddate: startOfDay.clone().add(15, 'weeks'),
      recurrence: WEEKLY,
    };
  };

  render() {
    const { index, initialValues, navigator, onSubmit } = this.props;
    return (
      <DecoratedAppointmentForm
        onSubmit={values => onSubmit(values, index)}
        initialValues={{
          ...this._getInitialValues(moment()),
          ...initialValues,
        }}
        onRecurrencePress={input =>
          navigator.push({
            screen: RECURRENCE_PICKER,
            title: i18n.t('repeat'),
            passProps: {
              input,
            },
          })
        }
      />
    );
  }
}

const ConnectedAppointmentFormScreen: (
  props: OwnProps,
) => React$Element<any> = connect()(AppointmentFormScreen);

export default ConnectedAppointmentFormScreen;
