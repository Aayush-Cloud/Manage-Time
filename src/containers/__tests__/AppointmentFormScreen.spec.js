// @flow
import React from 'react';
import { shallow } from 'enzyme';
import navigatorMock from '../../lib/navigatorMock';
import { AppointmentFormScreen } from '../AppointmentFormScreen';

const props = {
  dispatch: jest.fn(),
  navigator: navigatorMock,
  onSubmit: jest.fn(),
};

describe('AppointmentFormScreen', () => {
  it('renders an AppointmentForm correctly', () => {
    const wrapper = shallow(<AppointmentFormScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
