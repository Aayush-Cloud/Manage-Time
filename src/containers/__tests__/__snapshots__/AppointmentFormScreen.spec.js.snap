// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentFormScreen renders an AppointmentForm correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ReduxForm
    initialValues={
        Object {
            "enddate": "2017-09-15T22:00:00.000Z",
            "endtime": "2017-06-03T21:00:00.000Z",
            "recurrence": "WEEKLY",
            "startdate": "2017-06-02T22:00:00.000Z",
            "starttime": "2017-06-03T20:00:00.000Z",
          }
    }
    onRecurrencePress={[Function]}
    onSubmit={[Function]}
/>,
  "nodes": Array [
    <ReduxForm
      initialValues={
            Object {
                  "enddate": "2017-09-15T22:00:00.000Z",
                  "endtime": "2017-06-03T21:00:00.000Z",
                  "recurrence": "WEEKLY",
                  "startdate": "2017-06-02T22:00:00.000Z",
                  "starttime": "2017-06-03T20:00:00.000Z",
                }
      }
      onRecurrencePress={[Function]}
      onSubmit={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AppointmentFormScreen
        dispatch={[Function]}
        navigator={
                Object {
                        "dismissAllModals": [Function],
                        "dismissLightBox": [Function],
                        "dismissModal": [Function],
                        "handleDeepLink": [Function],
                        "pop": [Function],
                        "popToRoot": [Function],
                        "push": [Function],
                        "resetTo": [Function],
                        "setButtons": [Function],
                        "setOnNavigatorEvent": [Function],
                        "setTitle": [Function],
                        "showLightBox": [Function],
                        "showModal": [Function],
                        "toggleNavBar": [Function],
                      }
        }
        onSubmit={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AppointmentFormScreen {
        "_getInitialValues": [Function],
        "_handleNavigatorEvent": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "dispatch": [Function],
          "navigator": Object {
            "dismissAllModals": [Function],
            "dismissLightBox": [Function],
            "dismissModal": [Function],
            "handleDeepLink": [Function],
            "pop": [Function],
            "popToRoot": [Function],
            "push": [Function],
            "resetTo": [Function],
            "setButtons": [Function],
            "setOnNavigatorEvent": [Function],
            "setTitle": [Function],
            "showLightBox": [Function],
            "showModal": [Function],
            "toggleNavBar": [Function],
          },
          "onSubmit": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ReduxForm
          initialValues={
                    Object {
                              "enddate": "2017-09-15T22:00:00.000Z",
                              "endtime": "2017-06-03T21:00:00.000Z",
                              "recurrence": "WEEKLY",
                              "startdate": "2017-06-02T22:00:00.000Z",
                              "starttime": "2017-06-03T20:00:00.000Z",
                            }
          }
          onRecurrencePress={[Function]}
          onSubmit={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <ReduxForm
          initialValues={
                    Object {
                              "enddate": "2017-09-15T22:00:00.000Z",
                              "endtime": "2017-06-03T21:00:00.000Z",
                              "recurrence": "WEEKLY",
                              "startdate": "2017-06-02T22:00:00.000Z",
                              "starttime": "2017-06-03T20:00:00.000Z",
                            }
          }
          onRecurrencePress={[Function]}
          onSubmit={[Function]}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AppointmentFormScreen
    dispatch={[Function]}
    navigator={
        Object {
            "dismissAllModals": [Function],
            "dismissLightBox": [Function],
            "dismissModal": [Function],
            "handleDeepLink": [Function],
            "pop": [Function],
            "popToRoot": [Function],
            "push": [Function],
            "resetTo": [Function],
            "setButtons": [Function],
            "setOnNavigatorEvent": [Function],
            "setTitle": [Function],
            "showLightBox": [Function],
            "showModal": [Function],
            "toggleNavBar": [Function],
          }
    }
    onSubmit={[Function]}
/>,
}
`;
